cdap-client/src/main/java/co/cask/cdap/client/AuthorizationClient.java

```diff
@@ -210,11 +210,11 @@ private void executeExistingRolesRequest(Role role, HttpRequest request) throws
     }
   }
 
-  private HttpResponse executePrivilegeRequest(EntityId entityId, HttpRequest request) throws FeatureDisabledException,
+  private HttpResponse executePrivilegeRequest(HttpRequest request) throws FeatureDisabledException,
     UnauthenticatedException, IOException, NotFoundException, UnauthorizedException {
     HttpResponse httpResponse = doExecuteRequest(request, HttpURLConnection.HTTP_NOT_FOUND);
     if (HttpURLConnection.HTTP_NOT_FOUND == httpResponse.getResponseCode()) {
-      throw new NotFoundException(entityId);
+      throw new NotFoundException(httpResponse.getResponseBodyAsString());
```

Comment:
I wonder if we should do a similar thing in the other Clients also. (at least the methods that interact with more than a single entity).